system-tests/projects/downloads/cypress/integration/downloads_spec.ts

```diff
@@ -6,7 +6,27 @@ describe('downloads', () => {
   })
 
   it('handles csv file download', () => {
-    cy.get('[data-cy=download-csv]').click()
+    // HACK: monkey-patching Cypress.downloads.end to ensure the download has
+    // finished before asserting on its contents. this test was flakey b/c
+    // sometimes file was read before the browser had written its contents.
+    // consider creating events for downloads or some API, so that this can
+    // be done without hacks
```

Comment:
@ryanthemanuel Can you open a GitHub issue for this since it's a user facing problem that we could solve?

Reply:
Logged: https://github.com/cypress-io/cypress/issues/20550

Reply:
looks like we solved this with a long timeout in our example recipes: https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__download/cypress/integration/remote-download-spec.js#L33

Reply:
Yah, those recipes are a bit problematic in other ways as well.